lib/amazonka/src/Amazonka.hs

```diff
@@ -528,7 +528,7 @@ presign ::
 presign env time expires rq =
   Presign.presignWith
     (appEndo (getDual (Env.envOverride env)))
-    (Env.envAuth env)
+    (fromMaybe (error "Amazonka.presign: Env with no Auth") $ Env.envAuth env)
```

Comment:
Possibly not the best way to do this? But `Env`s with no credentials seem like they'll be incredibly rare...

Open to suggestions, including returning a `Maybe`, using a type-level tracker of whether we have auth info, or something else entirely.

Reply:
I'm not thrilled about making it partial - but I think doing it properly involves rewriting the `Auth` module and/or presigning, so if we really want this in 2.0 lets' just release it (with the appropriate caveats noted in Haddock comments) and plan to revisit.